script.js

```diff
@@ -1,4 +1,199 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+import { DECK } from "./deck.js";
+let cardDeck = DECK;
+const GAMESTATES = ["draw-cards", "stand-or-hit", "show-results"];
+let gameState = GAMESTATES[0];
+let results = {
+  dealer: {
+    score: 0,
+    cardsDrawn: [],
+    cardNames: [],
+    cardTotal: 0,
+  },
+  "Player 1": {
+    score: 0,
+    cardsDrawn: [],
+    cardNames: [],
+    cardTotal: 0,
+  },
+};
+const renderButtons = () => {
+  document
+    .querySelectorAll(".btn--conditional")
+    .forEach((e) => (e.style.display = "inline-block"));
+};
+const hideButtons = () => {
+  document
+    .querySelectorAll(".btn--conditional")
+    .forEach((e) => (e.style.display = "none"));
+};
+const shuffleArray = (array) => {
+  for (let i = array.length - 1; i > 0; i--) {
+    const j = Math.floor(Math.random() * (i + 1)); // FisherYates Algorithm
+    const temp = array[i];
+    array[i] = array[j];
+    array[j] = temp;
+  }
+};
+const drawCards = (numCards, player) => {
+  for (let _ = 0; _ < numCards; _++) {
+    // adds drawn cards to cardsDrawn
+    results[player].cardsDrawn.push(cardDeck.pop());
+  }
+  results[player].cardNames = results[player].cardsDrawn.map(
+    // get the card names
+    (card) => card.name
+  );
+  // compute total score
+  results[player].cardTotal = results[player].cardNames.reduce(
```

Comment:
It'd be a lot better to refactor the calculation of total score into a helper function as it seems like you'll probably need it somewhere else